### PR TITLE
Add exceptions for next-app

### PR DESCRIPTION
**Pull request by Josehower**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
 node_modules
 yarn-error.log
+yarn.lock
```
Comment (karlhorky) [inline .gitignore:3]: what's this?

File: templates/.eslintignore [hunk 2]
```diff
@@ -1,3 +1,9 @@
...
 *.config.js
-build
\ No newline at end of file
+build
+
+# Next.js app ignore paths
+.next
+next-env.d.ts
+pages/_app.js
+pages/api/hello.js
```
Comment (karlhorky) [inline templates/.eslintignore:9]: These shouldn't be here - we don't want to ignore linter errors in these for the students

File: templates/.eslintignore [hunk 3]
```diff
@@ -1,3 +1,9 @@
 .eslintrc.js
 *.config.js
-build
\ No newline at end of file
+build
+
+# Next.js app ignore paths
+.next
+next-env.d.ts
```
Comment (karlhorky) [inline templates/.eslintignore:7]: Not sure we need this either... so just check it to see why

**Comment by Josehower** [comment]
I think should be ok now, i'll leave this PR first and then i'll use the new version of eslint-config, to test the preflight on next app.

This way i make sure all is working smooth

File: yarn.lock [hunk 4]
```diff
@@ -0,0 +1,8 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
+# yarn lockfile v1
+
+
+"@types/node@^14.14.25":
+  version "14.14.25"
+  resolved "https://registry.yarnpkg.com/@types/node/-/node-14.14.25.tgz#15967a7b577ff81383f9b888aa6705d43fbbae93"
+  integrity sha512-EPpXLOVqDvisVxtlbvzfyqSsFeQxltFbluZNRndIb8tr9KiBnYNLzrc1N3pyKUCww2RNrfHDViqDWWE1LCJQtQ==
```
Comment (karlhorky) [inline yarn.lock:8]: please remove this file - we don't want `yarn.lock`

**Comment by karlhorky** [comment]
I think having the `.next` folder in `.eslintignore` is reasonable:

```
# Next.js build folder
.next
```

You can add this change back.

**Review by karlhorky — APPROVED** [review]

**Pull request merged**